client/app/bundles/course/users/components/buttons/PointManagementButtons.tsx

```diff
@@ -5,22 +5,24 @@ import {
   ExperiencePointsRowData,
 } from 'types/course/experiencePointsRecords';
 
+import {
+  deleteExperiencePointsRecord,
+  updateExperiencePointsRecord,
+} from 'course/experience-points/operations';
 import DeleteButton from 'lib/components/core/buttons/DeleteButton';
 import SaveButton from 'lib/components/core/buttons/SaveButton';
 import { useAppDispatch } from 'lib/hooks/store';
 import toast from 'lib/hooks/toast';
 
-import {
-  deleteExperiencePointsRecord,
-  updateExperiencePointsRecord,
-} from '../../operations';
-
 interface Props extends WrappedComponentProps {
   permissions: ExperiencePointsRecordPermissions;
   data: ExperiencePointsRowData;
   isDirty: boolean;
   isManuallyAwarded: boolean;
   handleSave: (newData: ExperiencePointsRowData) => void;
+  studentId: number;
+  isErrorInput: boolean;
+  isDownloading?: boolean;
```

Comment:
PointManagementButtons should not know the existence of download or whether there's an error. Instead, you can pass `disabled` as a prop and add the logic when calling this component. So you have something like `disabled = isErrorInput || isDownloading || isDirty`.

Better to use disabled as props and it should be non-optional.